ARCHITECTURE.md

```diff
@@ -0,0 +1,123 @@
+- GitHub API: The `package-backend` heavily relies on the GitHub API when created packages, updating packages, creating users, and verifying ownership of individual packages. The way we have attempted to work around this is by creating a Mock GitHub API Server that will be run with ExpressJS during Integration-Tests, but as expected API changes or many other factors can still cause our tests to lack here.
+- Database: The `package-backend` heavily relies on a Database to retrieve and store any type of data. We have luckily managed this aspect fairly well, by having our Integration-Test suite spin up a database on a Docker container locally to test against. By ensuring we keep the Database schema and versions accurate to what's available in production we are able to reliably test the majority of aspects in relation to the database. Other than testing instances of over-usage and scalability issues.
+
+### Error Handling
+
+Since any unrecoverable error should always be returned to the user, and never cause the server to crash, errors should always be handled, and caught around `try...catch` blocks, or to fail gracefully and return the output.
+
+Most modules generically should never handle errors themselves, and instead return the error state to the handler of their interaction, that is the `handlers/xxx_handler.js`. Since only the handler will have the `Request` and `Response` objects in order to properly handle an error and return it to the user as needed.
+
+The way we are able to accurately and consistently communicate an error to a higher level module, or to the handler of the interaction is through what we call here the `Server Status Object`. Which in short is an object with strict `key:value` pairs that can reliably pass information back to their root handler in order to communicate the exact error state, and further details if possible.
+
+The structure of the `Server Status Object` has much more thorough documentation within [`bubbled_errors.md`](/docs/reference/bubbled_errors.md), named as such since the `Server Status Object` allows errors to bubble up modules until they reach their root handler.
+
+Then once an error or failed state reaches the root handler, in most cases it will pass that along to the `common_handler.js` which contains the predetermined returns for each failure state, or in some special cases, the root handler can ignore the failed state and try to recover, or may even expect a failed error state to continue on. If the root handler is expecting a failed error state from the `Server Status Object` but receives a success state instead, it will then craft it's own failed state `Server Status Object` to pass to the `common_handler.js` for communicating to the user of the failed state.
```

Comment:
```suggestion
Then once an error or failed state reaches the root handler, in most cases it will pass that along to the `common_handler.js` which contains the predetermined returns for each failure state, or in some special cases, the root handler can ignore the failed state and try to recover, or may even expect a failed error state to continue on. If the root handler is expecting a failed error state from the `Server Status Object` but receives a success state instead, it will then craft its own failed state `Server Status Object` to pass to the `common_handler.js` for communicating to the user of the failed state.
```